### PR TITLE
Add atlas-based segmentation example

### PR DESCRIPTION
**Pull request by pchlap**

### CODE REVIEW

File: examples/atlas_segmentation.ipynb [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+    "    \"number_of_iterations\": 50,\n",
+    "    \"metric\": \"mean_squares\",\n",
+    "    \"optimiser\": \"gradient_descent\",\n",
+    "    \"verbose\": False\n",
+    "}\n",
+    "\n",
+    "user_settings[\"deformable_registration_settings\"] = {\n",
+    "    \"isotropic_resample\": True,\n",
+    "    \"resolution_staging\": [8],#,4,2],\n",
+    "    \"iteration_staging\": [50],#,50,25],\n",
```
Comment (rnfinnegan) [inline examples/atlas_segmentation.ipynb:341]: I don't think the optimiser will even get to 50 iterations at 8 mm resolution - so we could safely reduce this to something like 10. I suggest not worrying about it in this case, as it won't make much difference.

**Review by rnfinnegan — APPROVED** [review]
Great stuff.

**Comment by rnfinnegan** [comment]
Thanks for this @pchlap, looks really good. I like the comparison between SAS and MAS.

**Comment by pchlap** [comment]
Thanks @rnfinnegan, I'll clean up those lines where the iterations are specified with a portion of it commented out. Will merge once done :)

**Pull request merged**